src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.IntegrationAccount.tests.ps1

```diff
@@ -942,5 +942,327 @@ InModuleScope Arcus.Scripting.IntegrationAccount {
+                   Set-AzIntegrationAccountPartners -ResourceGroupName $resourceGroupName -Name $integrationAccountName -PartnerFilePath $partnerFilePath
+                } | Should -Throw -ExpectedMessage "At least one business identity must be supplied"
+
+                # Assert
+                Assert-VerifiableMock
+            }
+            It "Providing only the partnerFilePath to create an partner is OK" {
+                # Arrange
+                $subscriptionId = [guid]::NewGuid()
+                $resourceGroupName = "rg-infrastructure"
+                $integrationAccountName = "unexisting-integration-account"
+                $integrationAccountResourceId = "/subscriptions/$subscriptionId/resourceGroups/$resourceGroupName/providers/Microsoft.Logic/integrationAccounts/$integrationAccountName"
+                $partnerName = 'Dummy_New_Partner'
+                $partnerResourceId = "/subscriptions/$subscriptionId/resourceGroups/$resourceGroupName/providers/Microsoft.Logic/integrationAccounts/$integrationAccountName/partners/$partnerName"
+                $partnerFilePath = "$PSScriptRoot\Files\IntegrationAccount\Partners\$partnerName.json"
+
+                Mock Get-Content {
+                    return [PSCustomObject] @{
+                        name = 'Partner1';
+                        properties = [PSCustomObject] @{
+                            partnerType = 'B2B';
+                                content = [PSCustomObject] @{
+                                b2b = [PSCustomObject] @{
+                                    businessIdentities = [PSCustomObject] @{
+                                        qualifier = '1';
+                                        value = '12345';
+                                    }
+                                }
+                            }
+                        }
+                    } | ConvertTo-Json -Depth 5;
+                } -Verifiable
+                
+                Mock Get-AzIntegrationAccount {
+                    return [pscustomobject]@{ Id = $integrationAccountResourceId; Name = $integrationAccountName; Type = 'Microsoft.Logic/integrationAccounts'; Location = 'westeurope'; Sku = 'Free' }
+                } -Verifiable
+
+                Mock Get-AzIntegrationAccountPartner {
+                    return $null
+                } -Verifiable
+
+                Mock Set-AzIntegrationAccountPartner {
+                    return $null
+                }
+
+                Mock New-AzIntegrationAccountPartner {
+                    return [pscustomobject]@{ Id = $partnerResourceId; Name = $partnerName; Type = 'Microsoft.Logic/integrationAccounts/partners'; CreatedTime = [datetime]::UtcNow; ChangedTime = [datetime]::UtcNow }
+                }
+
+                # Act
+                { Set-AzIntegrationAccountPartners -ResourceGroupName $resourceGroupName -Name $integrationAccountName -PartnerFilePath $partnerFilePath } | 
+                    Should -Not -Throw
```

Comment:
I'm wondering if `-Not -Throw` has any added value here. If it does throw, it will let the test fail, just like if it weren't there.

Reply:
Did a quick test and indeed the behavior is the same with and without the statement.
I could remove it but this will need to be done for all unit tests (not just the partner ones), and maybe with the `Should -Not -Throw` it is a bit more descriptive?

Reply:
Yes, you may have a point in that. Let's leave it for now as-is. 👍